app/src/main/java/org/ole/planet/myplanet/services/SharedPrefManager.kt

```
@@ -143,11 +143,37 @@ class SharedPrefManager @Inject constructor(
         }
     }
 
+    private fun getSecurePref(key: String, default: String = ""): String {
+        val value = pref.getString(key, default) ?: default
+        if (value.startsWith("enc:")) {
+            return try {
+                org.ole.planet.myplanet.utils.SecurePrefs.decryptString(context, value.substring(4)) ?: ""
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)